src/libs/wds-status.test.ts

```diff
@@ -0,0 +1,492 @@
+  });
+
+  describe('if fetching Leo apps fails', () => {
+    it('returns unknown for all fields', async () => {
+      // Arrange
+      const mockAjax: DeepPartial<AjaxContract> = {
+        Apps: {
+          listAppsV2: jest.fn().mockRejectedValue(new Error('Something went wrong')),
+        },
+      };
+      asMockedFn(Ajax).mockReturnValue(mockAjax as AjaxContract);
+
+      // Act
+      const { result: hookReturnRef } = await renderHookInAct(() => useWdsStatus({ workspaceId }));
+
+      // Assert
+      expect(hookReturnRef.current.status).toEqual({
+        numApps: 'unknown',
+        appName: 'unknown',
+        appStatus: 'unknown',
+        proxyUrl: 'unknown',
+        wdsResponsive: 'unknown',
+        version: 'unknown',
+        wdsStatus: 'unresponsive',
+        wdsDbStatus: 'unknown',
+        wdsPingStatus: 'unknown',
+        wdsIamStatus: 'unknown',
+        defaultInstanceExists: 'unknown',
+      });
+    });
+  });
+
+  describe('if Leo apps are fetched successfully', () => {
+    describe('if no WDS app is present', () => {
+      it('returns number of apps and unknown for other fields', async () => {
+        // Arrange
+        const mockAjax: DeepPartial<AjaxContract> = {
+          Apps: {
+            listAppsV2: jest.fn().mockResolvedValue([]),
+          },
+        };
+        asMockedFn(Ajax).mockReturnValue(mockAjax as AjaxContract);
+
+        // Act
+        const { result: hookReturnRef } = await renderHookInAct(() => useWdsStatus({ workspaceId }));
+
+        // Assert
+        expect(hookReturnRef.current.status).toEqual({
+          numApps: '0',
+          appName: 'unknown',
+          appStatus: 'unknown',
+          proxyUrl: 'unknown',
+          wdsResponsive: 'unknown',
+          version: 'unknown',
+          wdsStatus: 'unresponsive',
+          wdsDbStatus: 'unknown',
+          wdsPingStatus: 'unknown',
+          wdsIamStatus: 'unknown',
+          defaultInstanceExists: 'unknown',
+        });
+      });
+    });
+  });
+
+  describe('if WDS app is present', () => {
+    const wdsApp: ListAppResponse = {
+      workspaceId: '6601fdbb-4b53-41da-87b2-81385f4a760e',
+      cloudContext: {
+        cloudProvider: 'AZURE',
+        cloudResource:
+          '0cb7a640-45a2-4ed6-be9f-63519f86e04b/ffd1069e-e34f-4d87-a8b8-44abfcba39af/mrg-terra-dev-previ-20230623095104',
+      },
+      kubernetesRuntimeConfig: {
+        numNodes: 1,
+        machineType: 'Standard_A2_v2',
+        autoscalingEnabled: false,
+      },
+      errors: [],
+      status: 'RUNNING',
+      proxyUrls: {
+        wds: 'https://lz34dd00bf3fdaa72f755eeea8f928bab7cd135043043d59d5.servicebus.windows.net/wds-6601fdbb-4b53-41da-87b2-81385f4a760e-6601fdbb-4b53-41da-87b2-81385f4a760e/',
+      },
+      appName: 'wds-6601fdbb-4b53-41da-87b2-81385f4a760e',
+      // @ts-expect-error Leo app types do not include WDS
+      appType: 'WDS',
+      // @ts-expect-error Leo app types do not reflect actual API responses.
```

Comment:
See thread in #4034: https://github.com/DataBiosphere/terra-ui/pull/4034#discussion_r1265648927